companyweb_base/models/companyweb_partner_model.py

```diff
@@ -0,0 +1,183 @@
+# Copyright 2021-2023 ACSONE SA/NV
+import logging
+from datetime import datetime
+from hashlib import sha1
+
+import zeep
+
+from odoo import fields, models
+from odoo.exceptions import ValidationError
+
+_logger = logging.getLogger(__name__)
+
+
+class CompanyWebPartner(models.Model):
+    _inherit = "res.partner"
+    cweb_lastupdate = fields.Datetime("Company Web Last Update", readonly=True)
+    cweb_name = fields.Char("CompanyWeb Name", readonly=True)
+    cweb_name_enable = fields.Boolean()
+    cweb_jur_form = fields.Char("CompanyWeb Juridical Form", readonly=True)
+    cweb_jur_form_enable = fields.Boolean()
+    cweb_street = fields.Char("CompanyWeb Street", readonly=True)
+    cweb_zip = fields.Char("CompanyWeb Postal code", readonly=True)
+    cweb_city = fields.Char("CompanyWeb City", readonly=True)
+    cweb_country = fields.Char("CompanyWeb Country", readonly=True)
+    cweb_address_enable = fields.Boolean()
+    cweb_creditLimit = fields.Float("CompanyWeb Credit limit", readonly=True)
+    cweb_creditLimit_enable = fields.Boolean()
+    cweb_startDate = fields.Date("CompanyWeb Start date", readonly=True)
+    cweb_startDate_enable = fields.Boolean()
+    cweb_endDate = fields.Date("CompanyWeb End date", readonly=True)
+    cweb_endDate_enable = fields.Boolean()
+    cweb_score = fields.Char("CompanyWeb Score", readonly=True)
+    cweb_score_enable = fields.Boolean()
+    cweb_image = fields.Char(readonly=True)
+    cweb_warnings = fields.Text("CompanyWeb Warnings", readonly=True)
+    cweb_warnings_enable = fields.Boolean()
+    cweb_url = fields.Char("CompanyWeb Detailed Report", readonly=True)
+    cweb_url_enable = fields.Boolean()
+    cweb_vat_liable = fields.Boolean("CompanyWeb Subject to VAT", readonly=True)
+    cweb_vat_liable_enable = fields.Boolean()
+    cweb_balance_year = fields.Char("CompanyWeb Balance Year", readonly=True)
+    cweb_balance_year_enable = fields.Boolean()
+    cweb_equityCapital = fields.Float("CompanyWeb Equity Capital", readonly=True)
+    cweb_addedValue = fields.Float("CompanyWeb Gross Margin (+/-)", readonly=True)
+    cweb_turnover = fields.Float("CompanyWeb Turnover", readonly=True)
+    cweb_result = fields.Float(
+        "CompanyWeb Fiscal Year Profit/Loss (+/-)", readonly=True
+    )
+    cweb_prefLang = fields.Char("CompanyWeb Preferred Language", readonly=True)
+    cweb_prefLang_enable = fields.Boolean()
+
+    def create_hash(self, login, password, secret):
+        today = datetime.now().strftime("%Y%m%d")
+        text = (today + login + password + secret).lower()
+        return sha1(text.encode("utf-8")).hexdigest()
+
+    def button_companyweb(self):
+        SERVICE_INTEGRATOR_ID = "acsone"
+        SERVICE_INTEGRATOR_SECRET = "ECAB8ACF-9AE1-4E90-BD0D-05A1F47A3FE9"
+        LOGIN = self.env.user.cweb_login
+        PASSWORD = self.env.user.cweb_password
```

Comment:
These are local variable, so the convention is to have them lowercase (uppercase if for constants)